### PR TITLE
Added id for Feauture.

### PR DESCRIPTION
**Pull request by GrumpyMan**
Many geojson files has an id for each Feature. I added support for the id field. I had to add references to two libraries to get the Equal function to work for this field since it is dynamic.

### CODE REVIEW

**Comment by hakenmt** [comment]
Id isn't part of the RFC, so I'm hesitant to include it in the base library. I'd probably prefer for users to extend the library to include implementation specific properties or methods.

**Comment by GrumpyMan** [comment]
Quoting from 3.2 in rfc7946

'If a Feature has a commonly used identifier, that identifier
      SHOULD be included as a member of the Feature object with the name
      "id", and the value of this member is either a JSON string or
      number.'

**Comment by hakenmt** [comment]
I somehow missed that both times I looked at the doc. I'll take another look at the PR. If the value was stored as a string in the object, but when serialized, was correctly interpreted as number or string, would that suffice?

**Comment by GrumpyMan** [comment]
It is a shortcut and those always introduce a risk.

I looked through the different files I parse now and I guess values in the input like:
- numbers without quotes - should be treated as a number but I think it could work also if they are treated as strings
- numbers with quotes - must be treated like strings. Think of leading zeroes "001" != "1". I have experienced similar cases.
- strings in quotes" - must be treated like strings.

To summarize I think it would work if all values are handled as strings as long as there are not geojson cases with a mix of numerical and string values for the id where it is possible to imagine cases where 1 != "1" and should be treated as different ids.

**Comment by hakenmt** [comment]
I created an alternative approach for including an id property in features that can be either a string or number without using dynamic. Still allows you to correctly compare id's for equality. Please check out the PR to see if it meets your requirements.

https://github.com/bamcis-io/GeoJSON/pull/23
